### PR TITLE
Feature/vdx 123/bls key resolver

### PR DESCRIPTION
**Pull request by maikel-maas**

### CODE REVIEW

File: packages/tsconfig.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     { "path": "waci-pex-qr-react" },
     { "path": "bls-kms-local" },
     { "path": "bls-did-provider-key" },
+    { "path": "bls-key-did-resolver" },
```
Comment (nklomp) [inline packages/tsconfig.json:17]: Can we rename it to bls-did-resolver-key, more inline with the providers?

Comment (maikel-maas) [reply packages/tsconfig.json]: Done!

File: packages/vc-handler-ld-local/README.md [hunk 2]
```diff
@@ -49,3 +49,114 @@ yarn test
...
 - `yarn fix` - runs `eslint --fix` as well as `prettier` to fix code style.
+
+### Request examples:
+
+#### Issue Verifiable Credential
+
+Ed25519:
+
+```shell
+POST http://localhost:7071/api/vdx-ivcs/v1/createVerifiableCredentialLDLocal
```
Comment (nklomp) [inline packages/vc-handler-ld-local/README.md:60]: This needs more context. This is using Veramo as a REST API, whilst most will not use it that way. So either make that known, or change the examples to just use the agent directly.

**Review by nklomp — CHANGES_REQUESTED** [review]
Some remarks.

I think we need to make a comment in the README why we created this module. What was missing from Veramo. Because probably we will want open a PR upstream to get it in Veramo itself

File: packages/bls-did-resolver-key/src/ed25519.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  const bytes = new Uint8Array(key.length + 2)
+  bytes[0] = 0xec
+  bytes[1] = 0x01
+  bytes.set(key, 2)
+  return `z${u8a.toString(bytes, 'base58btc')}`
+}
+export const keyToDidDoc = (pubKeyBytes: Uint8Array, fingerprint: string): DIDDocument => {
+  const did = `did:key:${fingerprint}`
+  const keyId = `${did}#${fingerprint}`
+  const x25519PubBytes = ed25519.convertPublicKeyToX25519(pubKeyBytes)
```
[the remark this answers is not in the document]
Comment (maikel-maas) [reply packages/bls-did-resolver-key/src/ed25519.ts:15]: The code comes from the original plugin.

File: packages/vc-handler-ld-local/README.md (hunk 2 above)
Comment (maikel-maas) [reply packages/vc-handler-ld-local/README.md:218]: Fixed!

File: packages/bls-did-resolver-key/src/ed25519.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (nklomp) [reply packages/bls-did-resolver-key/src/ed25519.ts:15]: It is just odd to be mixing them in the same code. But hey, we have more important things to worry about.

File: packages/vc-handler-ld-local/README.md [hunk 4]
```diff
@@ -50,7 +50,165 @@ There are other utility scripts that help with development.
...
+          store: new DIDStore(connection),
+          defaultProvider: 'did:key',
+          providers: {
+            'did:key': new BlsKeyDidProvider({
+              defaultKms: 'local',
+            }),
+            'did:web': new WebDIDProvider({
+              defaultKms: 'local',
+            }),
+            'did:factom': new FactomDIDProvider({
```
Comment (nklomp) [inline packages/vc-handler-ld-local/README.md:78]: Please remove the providers not being used in the example. Main reason is that I don't want our internal registrar links to end up all over our repos until VDX is in production

**Review by nklomp — CHANGES_REQUESTED** [review]
Single small fix needed for examples

File: packages/vc-handler-ld-local/README.md (hunk 4 above)
Comment (maikel-maas) [reply packages/vc-handler-ld-local/README.md]: Removed

**Review by nklomp — APPROVED** [review]

**Pull request merged**